src/pybind/mgr/dashboard/frontend/src/app/ceph/nfs/nfs-cluster/nfs-cluster.component.ts

```diff
@@ -0,0 +1,90 @@
+import { ActionLabelsI18n } from '~/app/shared/constants/app.constants';
+import { CdTableAction } from '~/app/shared/models/cd-table-action';
+import { CdTableColumn } from '~/app/shared/models/cd-table-column';
+import { CdTableSelection } from '~/app/shared/models/cd-table-selection';
+import { Permission } from '~/app/shared/models/permissions';
+import { AuthStorageService } from '~/app/shared/services/auth-storage.service';
+import { URLBuilderService } from '~/app/shared/services/url-builder.service';
+import { NFSClusterListConfig, NFSClusterTableModel } from '../models/nfs-cluster-config';
+import { OrchestratorStatus } from '~/app/shared/models/orchestrator.interface';
+import { OrchestratorService } from '~/app/shared/api/orchestrator.service';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { switchMap } from 'rxjs/operators';
+const BASE_URL = 'cephfs/nfs';
+@Component({
+  selector: 'cd-nfs-cluster',
+  templateUrl: './nfs-cluster.component.html',
+  styleUrls: ['./nfs-cluster.component.scss'],
+  providers: [{ provide: URLBuilderService, useValue: new URLBuilderService(BASE_URL) }]
+})
+export class NfsClusterComponent extends ListWithDetails implements OnInit {
+  @ViewChild('hostnameTpl', { static: true })
+  hostnameTpl: TemplateRef<any>;
+
+  @ViewChild('ipAddrTpl', { static: true })
+  ipAddrTpl: TemplateRef<any>;
+
+  @ViewChild('virtualIpTpl', { static: true })
+  virtualIpTpl: TemplateRef<any>;
+
+  columns: CdTableColumn[] = [];
+  selection: CdTableSelection = new CdTableSelection();
+  tableActions: CdTableAction[] = [];
+  data1: NFSClusterTableModel[];
+  permission: Permission;
+  orchStatus: OrchestratorStatus;
+  clusters$: Observable<Object>;
+  subject = new BehaviorSubject<NFSClusterListConfig[]>([]);
+  constructor(
+    public actionLabels: ActionLabelsI18n,
+    protected ngZone: NgZone,
+    private authStorageService: AuthStorageService,
+    private nfsService: NfsService,
+    private orchService: OrchestratorService
+  ) {
+    super();
+  }
+  ngOnInit(): void {
+    this.orchService.status().subscribe((status: OrchestratorStatus) => {
+      this.orchStatus = status;
+    });
+    this.permission = this.authStorageService.getPermissions().nfs;
+    this.loadData();
+    this.columns = [
+      {
+        name: $localize`Cluster Name`,
+        prop: 'name',
+        flexGrow: 1
+      },
+      {
+        name: $localize`Hostnames`,
+        prop: 'backend',
+        flexGrow: 2,
+        cellTemplate: this.hostnameTpl
+      },
+      {
+        name: $localize`IP Address`,
+        prop: 'backend',
+        flexGrow: 2,
+        cellTemplate: this.ipAddrTpl
+      },
+      {
+        name: $localize`Virtual IP Address`,
+        prop: 'virtual_ip',
+        flexGrow: 1,
+        cellTemplate: this.virtualIpTpl
+      }
+    ];
+  }
+
+  loadData() {
+    this.clusters$ = this.subject.pipe(switchMap(() => this.nfsService.nfsClusterList()));
```

Comment:
you can do this in ngOninit()
```
    this.clusters$ = this.subject.pipe(switchMap(() => this.nfsService.nfsClusterList()));
```
and then in the loadData call you just have to get the next of subject
```suggestion
   this.subject.next([]);
```

Reply:
updated